src/components/settings/SettingsSSOTab/hooks.js

```diff
@@ -75,14 +86,17 @@ const useIdpState = () => {
       // but we need to update this support the case when the correct sso config must be updated
       setProviderConfig(response.data);
 
-      // also get samlproviderdata updated from remote metadata url
-      const providerdataResponse = await updateSamlProviderData({
-        enterpriseId,
-        metadataURL,
-        entityID,
+      // If the user already has a provider data entry with a different entityID, remove it before
+      // creating a new one
+      existingProviderData.forEach(async (idpData) => {
```

Comment:
deleting existing provider configs that have different entity id's than the current one the admin is trying to use